### ISSUE TITLE
Unable to set message property when running in Ubuntu container: MQRC_HCONN_ERROR

### ISSUE THREAD

**Issue by KYDronePilot** [dsuch/pymqi#147]
I am having issues trying to set a message property when running pymqi in an Ubuntu Docker container. Here is the gist of the code that gets executed:

```python
handle = pymqi.MessageHandle(qmgr=queue_manager)
handle.properties.set(b'some_key', 'some_val')
```

and the relevant part of the traceback:
```
File "/usr/local/lib/python3.7/dist-packages/pymqi/__init__.py", line 2374, in set
    raise MQMIError(comp_code, comp_reason) pymqi.MQMIError: MQI Error. Comp: 2, Reason 2018: FAILED: MQRC_HCONN_ERROR
```

When running on Windows, however, the error does not occur. Here are the specifications for the Windows machine and the Ubuntu container:

**Windows machine**:
 - OS: Windows 10 (1903)
 - MQ: 9.0.4.0 Win64 client
 - Python: 3.7.4
 - pymqi: 1.9.2 (built from commit c3a0f19dc991a7b26e34f0e3142ec99e36673519)

**Ubuntu container**:
 - Version: 19.04
 - MQ: 9.1.3.0 Redist LinuxX64 client (same issue occurred with older versions of client)
 - Python: 3.7.3
 - pymqi: 1.9.3

I was able to get MQ traces for both machines when trying to set a message property:

**Windows (working)**:
```
...
...
...
00088986 11:44:50.995367  13728.4           :       !! - Thread stack (from mqe.dll)
00088987 11:44:50.995436  13728.4           :       !! - -> MQCRTMH
00088988 11:44:50.995443  13728.4           :       !! - -> tmzstMQCRTMH
00088989 11:44:50.995449  13728.4           :       !! - -> reqInitClientThread
0008898A 11:44:50.995454  13728.4           :       !! - -> xcsInitialize
0008898B 11:44:50.995459  13728.4           :       !! - -> xcsInitGlobalSecurityData
0008898C 11:44:50.995464  13728.4           :       ------{  xcsInitGlobalSecurityData
0008898D 11:44:50.995469  13728.4           :       ------}  xcsInitGlobalSecurityData (rc=OK)
0008898E 11:44:50.995474  13728.4           :       ------{  xcsCreateNTSecurityAtts
0008898F 11:44:50.995481  13728.4           :       ------}  xcsCreateNTSecurityAtts (rc=OK)
00088990 11:44:50.995489  13728.4           :       ------{  xrnInitialise
00088991 11:44:50.995494  13728.4           :       -------{  xrnGetAnchor
00088992 11:44:50.995498  13728.4           :       -------}  xrnGetAnchor (rc=OK)
00088993 11:44:50.995502  13728.4           :       Anchor eye xrnA flags 0x39
00088994 11:44:50.995508  13728.4           :       ------}  xrnInitialise (rc=OK)
00088995 11:44:50.995514  13728.4           :       xcsInitialize  xihThreadInitsCount: 1, xihPrivateInitsCount: 0, xihClientInitsCount: 1, xihPreInitialized: FALSE, xihAllInitsCount: 6, pCS: XXXXXXXXXXXXXXXX
00088996 11:44:50.995520  13728.4           :       ------{  xcsReleaseThreadMutexSem
00088997 11:44:50.995525  13728.4           :       ------}  xcsReleaseThreadMutexSem (rc=OK)
00088998 11:44:50.995528  13728.4           :       -----}  xcsInitialize (rc=OK)
00088999 11:44:50.995532  13728.4           :       -----{  xcsGetEnvironmentString
0008899A 11:44:50.995539  13728.4           :       xcsGetEnvironmentString[MQNOWORK] = NULL
0008899B 11:44:50.995545  13728.4           :       -----}! xcsGetEnvironmentString (rc=xecE_E_ENV_VAR_NOT_FOUND)
0008899C 11:44:50.995553  13728.4           :       -----{  xcsGetMemFn
0008899D 11:44:50.995559  13728.4           :       component:20 function:397 length:2408 options:0 cbmindex:-1 *pointer:XXXXXXXXXXXXXXXX
0008899E 11:44:50.995565  13728.4           :       -----}  xcsGetMemFn (rc=OK)
0008899F 11:44:50.995598  13728.4           :       -----{  cccCSRegister
000889A0 11:44:50.995603  13728.4           :       ------{  xcsGetEnvironmentString
000889A1 11:44:50.995630  13728.4           :       xcsGetEnvironmentString[AMQ_FFST_REMOTE] = NULL
000889A2 11:44:50.995636  13728.4           :       ------}! xcsGetEnvironmentString (rc=xecE_E_ENV_VAR_NOT_FOUND)
000889A3 11:44:50.995647  13728.4           :       ------{  xcsRegComp
000889A4 11:44:50.995652  13728.4           :       comp:20 Count:0
000889A5 11:44:50.995662  13728.4           :       ------}  xcsRegComp (rc=OK)
000889A6 11:44:50.995667  13728.4           :       -----}  cccCSRegister (rc=OK)
000889A7 11:44:50.995672  13728.4           :       -----{  xcsRequestThreadMutexSem
000889A8 11:44:50.995680  13728.4           :       -----}  xcsRequestThreadMutexSem (rc=OK)
000889A9 11:44:50.995686  13728.4           :       -----{  xcsReleaseThreadMutexSem
000889AA 11:44:50.995691  13728.4           :       -----}  xcsReleaseThreadMutexSem (rc=OK)
000889AB 11:44:50.995695  13728.4           :       -----{  xcsRegisterDestructor
000889AC 11:44:50.995701  13728.4           :       regType(1) destructorType(2) routine(00007FF8662ED820) rmId(0) arg(XXXXXXXXXXXXXXXX)
000889AD 11:44:50.995708  13728.4           :       !! - Not retaining destructor since library is self-contained
000889AE 11:44:50.995712  13728.4           :       ------{  xcsGetMemFn
000889AF 11:44:50.995718  13728.4           :       component:23 function:258 length:48 options:0 cbmindex:-1 *pointer:XXXXXXXXXXXXXXXX
000889B0 11:44:50.995724  13728.4           :       ------}  xcsGetMemFn (rc=OK)
000889B1 11:44:50.995728  13728.4           :       -----}  xcsRegisterDestructor (rc=OK)
000889B2 11:44:50.995732  13728.4           :       -----{  cccGetConfig
000889B3 11:44:50.995736  13728.4           :       ------{  xcsRequestThreadMutexSem
000889B4 11:44:50.995741  13728.4           :       ------}  xcsRequestThreadMutexSem (rc=OK)
000889B5 11:44:50.995745  13728.4           :       ------{  xcsReleaseThreadMutexSem
000889B6 11:44:50.995763  13728.4           :       ------}  xcsReleaseThreadMutexSem (rc=OK)
000889B7 11:44:50.995767  13728.4           :       Configuration Block
000889B8 11:44:50.995792  13728.4           :       Data:-
                                                    <<<20 lines omitted>>>
000889B9 11:44:50.995931  13728.4           :       -----}  cccGetConfig (rc=OK)
000889BA 11:44:50.995935  13728.4           :       ----}  reqInitClientThread (rc=OK)
000889BB 11:44:50.995939  13728.4           :       ----{  MQCRTMH
000889BC 11:44:50.995943  13728.4           :       -----{  reqGetConn
000889BD 11:44:50.995947  13728.4           :       ------{  zutLookupGet
000889BE 11:44:50.995952  13728.4           :       Data: 0xXXXXXXXX
000889BF 11:44:50.995961  13728.4           :       -------{  zutLookupLock
000889C0 11:44:50.995965  13728.4           :       -------}  zutLookupLock (rc=OK)
000889C1 11:44:50.995969  13728.4           :       -------{  zutLookupUnlock
000889C2 11:44:50.995973  13728.4           :       -------}  zutLookupUnlock (rc=OK)
000889C3 11:44:50.995977  13728.4           :       ------}  zutLookupGet (rc=OK)
000889C4 11:44:50.995995  13728.4           :       ------{  xcsRequestThreadMutexSem
000889C5 11:44:50.995999  13728.4           :       ------}  xcsRequestThreadMutexSem (rc=OK)
000889C6 11:44:50.996022  13728.4      RSESS:000003 -----}  reqGetConn (rc=OK)
000889C7 11:44:50.996026  13728.4      RSESS:000003 -----{  zstMQCRTMH_CheckParms
000889C8 11:44:50.996030  13728.4      RSESS:000003 ------{  xcsCheckPointer
000889C9 11:44:50.996034  13728.4      RSESS:000003 ------}  xcsCheckPointer (rc=OK)
000889CA 11:44:50.996038  13728.4      RSESS:000003 ------{  xcsCheckPointer
000889CB 11:44:50.996041  13728.4      RSESS:000003 ------}  xcsCheckPointer (rc=OK)
000889CC 11:44:50.996045  13728.4      RSESS:000003 -----}  zstMQCRTMH_CheckParms (rc=OK)
000889CD 11:44:50.996049  13728.4      RSESS:000003 !! - ___________
000889CE 11:44:50.996053  13728.4      RSESS:000003 !! - MQCRTMH >>
000889CF 11:44:50.996057  13728.4      RSESS:000003 !! - Hconn:
000889D0 11:44:50.996060  13728.4      RSESS:000003 Data:-
000889D0 11:44:50.996060  13728.4      RSESS:000003  0x000000FA 024CA750 09 00 00 01                                     : ....
000889D1 11:44:50.996068  13728.4      RSESS:000003 !! - CrtMsgHOpts:
000889D2 11:44:50.996071  13728.4      RSESS:000003 Data:-
000889D2 11:44:50.996071  13728.4      RSESS:000003  0x000002A8 712D1AD0 43 4D 48 4F 01 00 00 00 00 00 00 00             : CMHO........
000889D3 11:44:50.996079  13728.4      RSESS:000003 !! - Hmsg:
000889D4 11:44:50.996082  13728.4      RSESS:000003 Data:-
000889D4 11:44:50.996082  13728.4      RSESS:000003  0x000000FA XXXXXXXX FF FF FF FF FF FF FF FF                         : XXXXXXXX
000889D5 11:44:50.996091  13728.4      RSESS:000003 !! - Compcode      : Output Parm
000889D6 11:44:50.996095  13728.4      RSESS:000003 !! - Reason        : Output Parm
000889D7 11:44:50.996112  13728.4      RSESS:000003 -----{  zstCreateMessageHandle
000889D8 11:44:50.996115  13728.4      RSESS:000003 ------{  xcsGetMemFn
000889D9 11:44:50.996139  13728.4      RSESS:000003 component:32 function:429 length:632 options:0 cbmindex:-1 *pointer:XXXXXXXXXXXXXXXX
000889DA 11:44:50.996148  13728.4      RSESS:000003 ------}  xcsGetMemFn (rc=OK)
000889DB 11:44:50.996152  13728.4      RSESS:000003 ------{  xcsGetMemFn
000889DC 11:44:50.996224  13728.4      RSESS:000003 component:32 function:429 length:4096 options:0 cbmindex:-1 *pointer:XXXXXXXXXXXXXXXX
000889DD 11:44:50.996232  13728.4      RSESS:000003 ------}  xcsGetMemFn (rc=OK)
000889DE 11:44:50.996236  13728.4      RSESS:000003 MsgHandleCount: 1
000889DF 11:44:50.996244  13728.4      RSESS:000003 Data: 0xXXXXXXXX 0xXXXXXXXX
000889E0 11:44:50.996251  13728.4      RSESS:000003 -----}  zstCreateMessageHandle (rc=OK)
000889E1 11:44:50.996254  13728.4      RSESS:000003 !! - ___________
000889E2 11:44:50.996258  13728.4      RSESS:000003 !! - MQCRTMH <<
000889E3 11:44:50.996262  13728.4      RSESS:000003 !! - Hconn         : Input  Parm
000889E4 11:44:50.996266  13728.4      RSESS:000003 !! - CrtMsgHOpts:
000889E5 11:44:50.996269  13728.4      RSESS:000003 Data:-
000889E5 11:44:50.996269  13728.4      RSESS:000003  0x000002A8 712D1AD0 43 4D 48 4F 01 00 00 00 00 00 00 00             : CMHO........
000889E6 11:44:50.996277  13728.4      RSESS:000003 !! - Hmsg:
000889E7 11:44:50.996280  13728.4      RSESS:000003 Data:-
000889E7 11:44:50.996280  13728.4      RSESS:000003  0x000000FA XXXXXXXX XX XX XX XX XX XX XX XX                         : XXXXXXXX
000889E8 11:44:50.996288  13728.4      RSESS:000003 !! - Compcode:
000889E9 11:44:50.996292  13728.4      RSESS:000003 Data:-
000889E9 11:44:50.996292  13728.4      RSESS:000003  0x000000FA 024CA868 00 00 00 00                                     : ....
000889EA 11:44:50.996299  13728.4      RSESS:000003 !! - Reason:
000889EB 11:44:50.996302  13728.4      RSESS:000003 Data:-
000889EB 11:44:50.996302  13728.4      RSESS:000003  0x000000FA 024CA860 00 00 00 00                                     : ....
000889EC 11:44:50.996310  13728.4      RSESS:000003 -----{  reqReleaseConn
000889ED 11:44:50.996314  13728.4      RSESS:000003 ------{  zutLookupRelease
000889EE 11:44:50.996318  13728.4      RSESS:000003 ------}  zutLookupRelease (rc=OK)
000889EF 11:44:50.996322  13728.4      RSESS:000003 ------{  xcsQueryThreadMutexSem
000889F0 11:44:50.996328  13728.4      RSESS:000003 hmtx is XXXXXXXXXXXXXXXX
000889F1 11:44:50.996335  13728.4      RSESS:000003 ------}! xcsQueryThreadMutexSem (rc=Unknown(1))
000889F2 11:44:50.996344  13728.4      RSESS:000003 ------{  xcsReleaseThreadMutexSem
000889F3 11:44:50.996349  13728.4      RSESS:000003 ------}  xcsReleaseThreadMutexSem (rc=OK)
000889F4 11:44:50.996352  13728.4      RSESS:000003 -----}  reqReleaseConn (rc=OK)
000889F5 11:44:50.996356  13728.4      RSESS:000003 ----}  MQCRTMH (rc=OK)
000889F6 11:44:50.996360  13728.4      RSESS:000003 ---}  tmzstMQCRTMH (rc=OK)
000889F7 11:44:50.996364  13728.4      RSESS:000003 --}  MQCRTMH (rc=OK)
000889F8 11:44:50.996454  13728.4      RSESS:000003 --{  MQSETMP
000889F9 11:44:50.996460  13728.4      RSESS:000003 ---{  tmzstMQSETMP
000889FA 11:44:50.996464  13728.4      RSESS:000003 !! - Passing function directly to real mqm
000889FB 11:44:50.996469  13728.4      RSESS:000003 ----{  MQSETMP
000889FC 11:44:50.996472  13728.4      RSESS:000003 -----{  reqGetConn
000889FD 11:44:50.996476  13728.4      RSESS:000003 ------{  zutLookupGet
000889FE 11:44:50.996481  13728.4      RSESS:000003 Data: 0xXXXXXXXX
000889FF 11:44:50.996488  13728.4      RSESS:000003 -------{  zutLookupLock
00088A00 11:44:50.996492  13728.4      RSESS:000003 -------}  zutLookupLock (rc=OK)
00088A01 11:44:50.996495  13728.4      RSESS:000003 -------{  zutLookupUnlock
00088A02 11:44:50.996499  13728.4      RSESS:000003 -------}  zutLookupUnlock (rc=OK)
00088A03 11:44:50.996503  13728.4      RSESS:000003 ------}  zutLookupGet (rc=OK)
00088A04 11:44:50.996507  13728.4      RSESS:000003 ------{  xcsRequestThreadMutexSem
00088A05 11:44:50.996512  13728.4      RSESS:000003 ------}  xcsRequestThreadMutexSem (rc=OK)
00088A06 11:44:50.996516  13728.4      RSESS:000003 -----}  reqGetConn (rc=OK)
00088A07 11:44:50.996539  13728.4      RSESS:000003 -----{  zstMQSETMP_CheckParms
00088A08 11:44:50.996544  13728.4      RSESS:000003 ------{  xcsCheckPointer
00088A09 11:44:50.996548  13728.4      RSESS:000003 ------}  xcsCheckPointer (rc=OK)
00088A0A 11:44:50.996552  13728.4      RSESS:000003 ------{  zutCheckValidPropertyName
00088A0B 11:44:50.996556  13728.4      RSESS:000003 -------{  xcsCheckPointer
00088A0C 11:44:50.996560  13728.4      RSESS:000003 -------}  xcsCheckPointer (rc=OK)
00088A0D 11:44:50.996564  13728.4      RSESS:000003 -------{  xcsCheckPointer
00088A0E 11:44:50.996568  13728.4      RSESS:000003 -------}  xcsCheckPointer (rc=OK)
00088A0F 11:44:50.996572  13728.4      RSESS:000003 ------}  zutCheckValidPropertyName (rc=OK)
00088A10 11:44:50.996576  13728.4      RSESS:000003 ------{  xcsCheckPointer
00088A11 11:44:50.996579  13728.4      RSESS:000003 ------}  xcsCheckPointer (rc=OK)
00088A12 11:44:50.996583  13728.4      RSESS:000003 ------{  xcsCheckNullTermString
00088A13 11:44:50.996587  13728.4      RSESS:000003 ------}  xcsCheckNullTermString (rc=OK)
00088A14 11:44:50.996591  13728.4      RSESS:000003 -----}  zstMQSETMP_CheckParms (rc=OK)
00088A15 11:44:50.996595  13728.4      RSESS:000003 !! - ___________
00088A16 11:44:50.996598  13728.4      RSESS:000003 !! - MQSETMP >>
...
...
...
```

**Ubuntu container (not working)**:
```
...
...
...
*11:21:43.912377        8.1      RSESS:000001      Thread stack (from libmqe_r.so)
 11:21:43.913198        8.1      RSESS:000001      -> MQCRTMH
 11:21:43.913208        8.1      RSESS:000001      -> MQCRTMH
 11:21:43.913212        8.1      RSESS:000001 -{  MQCRTMH
 11:21:43.913215        8.1      RSESS:000001 --{  reqGetConn
 11:21:43.913218        8.1      RSESS:000001 ---{  zutLookupGet
 11:21:43.913224        8.1      RSESS:000001      Data: 0x*******1
 11:21:43.913226        8.1      RSESS:000001 ----{  zutLookupLock
 11:21:43.913229        8.1      RSESS:000001 ----}  zutLookupLock rc=OK FunctionTime=3
 11:21:43.913231        8.1      RSESS:000001 ----{  zutLookupUnlock
 11:21:43.913234        8.1      RSESS:000001 ----}  zutLookupUnlock rc=OK FunctionTime=3
 11:21:43.913236        8.1      RSESS:000001 ---}  zutLookupGet rc=OK FunctionTime=18
 11:21:43.913239        8.1      RSESS:000001 ---{  xcsRequestThreadMutexSem
 11:21:43.913245        8.1      RSESS:000001      hmtx is 0x*******2, Timeout is -1
 11:21:43.913247        8.1      RSESS:000001 ---}  xcsRequestThreadMutexSem rc=OK FunctionTime=8
 11:21:43.913250        8.1      RSESS:000001 --}  reqGetConn rc=OK FunctionTime=35
 11:21:43.913253        8.1      RSESS:000001 --{  zstMQCRTMH_CheckParms
 11:21:43.913255        8.1      RSESS:000001 ---{  xcsCheckPointer
 11:21:43.913258        8.1      RSESS:000001 ---}  xcsCheckPointer rc=OK FunctionTime=3
 11:21:43.913261        8.1      RSESS:000001 ---{  xcsCheckPointer
 11:21:43.913263        8.1      RSESS:000001 ---}  xcsCheckPointer rc=OK FunctionTime=2
 11:21:43.913266        8.1      RSESS:000001 --}  zstMQCRTMH_CheckParms rc=OK FunctionTime=13
 11:21:43.913269        8.1      RSESS:000001      ___________
 11:21:43.913271        8.1      RSESS:000001      MQCRTMH >>
 11:21:43.913274        8.1      RSESS:000001      Hconn:
 11:21:43.913276        8.1      RSESS:000001        0x0000:  0500007e                             |...~
   |
 11:21:43.913279        8.1      RSESS:000001      CrtMsgHOpts:
 11:21:43.913281        8.1      RSESS:000001        0x0000:  434d484f 01000000 00000000           |CMHO........ 
   |
 11:21:43.913284        8.1      RSESS:000001      Hmsg:
 11:21:43.913286        8.1      RSESS:000001        0x0000:  00000000 ffffffff                    |........     
   |
 11:21:43.913288        8.1      RSESS:000001      Compcode      : Output Parm
 11:21:43.913291        8.1      RSESS:000001      Reason        : Output Parm
 11:21:43.913303        8.1      RSESS:000001 --{  zstCreateMessageHandle
 11:21:43.913305        8.1      RSESS:000001 ---{  xcsGetMemFn
 11:21:43.913309        8.1      RSESS:000001      Data: 0x00000000 0x*********3
 11:21:43.913312        8.1      RSESS:000001      component:32 function:429 length:632 options:0 cbmindex:4 *pointer:0x*******4
 11:21:43.913314        8.1      RSESS:000001 ---}  xcsGetMemFn rc=OK FunctionTime=9
 11:21:43.913317        8.1      RSESS:000001 ---{  xcsGetMemFn
 11:21:43.913320        8.1      RSESS:000001      Data: 0x00000000 0x********6
 11:21:43.913323        8.1      RSESS:000001      component:32 function:429 length:4096 options:0 cbmindex:-1 *pointer:0x*******5
 11:21:43.913325        8.1      RSESS:000001 ---}  xcsGetMemFn rc=OK FunctionTime=8
 11:21:43.913327        8.1      RSESS:000001      MsgHandleCount: 2
 11:21:43.913329        8.1      RSESS:000001      Data: 0x00000000 0x*********3
 11:21:43.913331        8.1      RSESS:000001 --}  zstCreateMessageHandle rc=OK FunctionTime=28
 11:21:43.913333        8.1      RSESS:000001      ___________
 11:21:43.913335        8.1      RSESS:000001      MQCRTMH <<
 11:21:43.913337        8.1      RSESS:000001      Hconn         : Input  Parm
 11:21:43.913339        8.1      RSESS:000001      CrtMsgHOpts:
 11:21:43.913341        8.1      RSESS:000001        0x0000:  434d484f 01000000 00000000           |CMHO........ 
   |
 11:21:43.913343        8.1      RSESS:000001      Hmsg:
 11:21:43.913345        8.1      RSESS:000001        0x0000:  10987202 00000000                    |..r.....     
   |
 11:21:43.913347        8.1      RSESS:000001      Compcode:
 11:21:43.913349        8.1      RSESS:000001        0x0000:  00000000                             |....
   |
 11:21:43.913351        8.1      RSESS:000001      Reason:
 11:21:43.913353        8.1      RSESS:000001        0x0000:  00000000                             |....
   |
 11:21:43.913355        8.1      RSESS:000001 --{  reqReleaseConn
 11:21:43.913357        8.1      RSESS:000001 ---{  xcsRequestThreadMutexSem
 11:21:43.913360        8.1      RSESS:000001      hmtx is 0x*******7, Timeout is -1
 11:21:43.913362        8.1      RSESS:000001 ---}  xcsRequestThreadMutexSem rc=OK FunctionTime=5
 11:21:43.913364        8.1      RSESS:000001 ---{  zutLookupRelease
 11:21:43.913366        8.1      RSESS:000001 ---}  zutLookupRelease rc=OK FunctionTime=2
 11:21:43.913368        8.1      RSESS:000001 ---{  xcsQueryThreadMutexSem
 11:21:43.913370        8.1      RSESS:000001      hmtx is 0x*******2
 11:21:43.913372        8.1      RSESS:000001 ---}! xcsQueryThreadMutexSem rc=Unknown(1) FunctionTime=4
 11:21:43.913374        8.1      RSESS:000001 ---{  xcsReleaseThreadMutexSem
 11:21:43.913376        8.1      RSESS:000001      hmtx is 0x*******2
 11:21:43.913378        8.1      RSESS:000001 ---}  xcsReleaseThreadMutexSem rc=OK FunctionTime=4
 11:21:43.913380        8.1      RSESS:000001 ---{  xcsReleaseThreadMutexSem
 11:21:43.913382        8.1      RSESS:000001      hmtx is 0x*******7
 11:21:43.913384        8.1      RSESS:000001 ---}  xcsReleaseThreadMutexSem rc=OK FunctionTime=4
 11:21:43.913386        8.1      RSESS:000001 --}  reqReleaseConn rc=OK FunctionTime=31
 11:21:43.913388        8.1      RSESS:000001 -}  MQCRTMH rc=OK FunctionTime=176
 11:21:43.913390        8.1      RSESS:000001 }  MQCRTMH rc=OK FunctionTime=-1
 11:21:43.913449        8.1      RSESS:000001 {  MQSETMP
 11:21:43.913453        8.1      RSESS:000001 -{  MQSETMP
 11:21:43.913455        8.1      RSESS:000001 --{  reqGetConn
 11:21:43.913457        8.1      RSESS:000001 --}! reqGetConn rc=MQRC_HCONN_ERROR FunctionTime=2
 11:21:43.913459        8.1      RSESS:000001 -}! MQSETMP rc=MQRC_HCONN_ERROR FunctionTime=6
 11:21:43.913461        8.1      RSESS:000001 }! MQSETMP rc=MQRC_HCONN_ERROR FunctionTime=12
...
...
...
```

Any idea what is causing the issue and what could be done to fix it?

**Comment by JoshMcMQ** [comment]
Does it work on Ubuntu with 1.9.2? Does it fail on windows with 1.9.3?

**Comment by JoshMcMQ** [comment]
@SeyfSV this may also be related to https://github.com/dsuch/pymqi/commit/8952d5330394ab2057b590ba8c29a7920e06a2a8.

**Comment by KYDronePilot** [comment]
Running 1.9.2 on Ubuntu gives a new error:

```python
Traceback (most recent call last):
    ...
    ...
    File "/***.py", line ***, in ***
        handle = pymqi.MessageHandle(qmgr=manager)
    File "/usr/local/lib/python3.7/dist-packages/pymqi/__init__.py", line 2390, in __init__
        cmho.pack())
pymqe.error: MQCMHO wrong size. Given: 140187732541452, expected 12
```

1.9.3 appears to work fine on Windows, but it may be because I have an older MQ client installed. I am working on getting a Windows machine setup with the same version of MQ as the Ubuntu container and see if that makes a difference.

**Comment by chughts** [comment]
This issue (and definitely the last post) stink of the default structure versions set in pymqi 1.9.3. 

The last issue will be cured if you amend the line where you construct the CD structure with a version. 

eg.
`cd = pymqi.CD(Version=pymqi.CMQXC.MQCD_VERSION_11)`

**Comment by SeyfSV** [comment]
Hi @KYDronePilot,

The issue is strange because MessageHandle class instances lost connection handle, as I think, or it think that it lost it.
Please check `qmgr.get_handle()` before calling `pymqi.MessageHandle()`

Can you provide fully functional part of code that reproduce issue?
As alternative,  can you run test `code/tests/test_message_property.py` in container?

**Comment by KYDronePilot** [comment]
@chughts I tried adding the version arg to the CD constructor, but it still fails with the same error (another really large number). Side note: these pymqi-1.9.2 Ubuntu tests I have been doing (with and without the CD version arg) were on MQ client-9.0.4.0, since I could not get MQ client-9.1.3.0 working with pymqi-1.9.3.

@SeyfSV I was able to call `qmgr.get_handle()` both before and after calling `pymqi.MessageHandle()` without any errors.

I was able to construct a snippet that demonstrates the issue. The following runs fine on the Windows setup, but fails on the last line on the Ubuntu container:

```python
import pymqi


broker = 'host.com'
name = 'name'
channelName = 'some.channel'

qm = pymqi.QueueManager(None)
cd = pymqi.CD()
cd.ChannelName = channelName.encode()
cd.ConnectionName = broker.encode()
cd.ChannelType = pymqi.CMQC.MQCHT_CLNTCONN
cd.TransportType = pymqi.CMQC.MQXPT_TCP
connectOptions = pymqi.CMQC.MQCNO_HANDLE_SHARE_BLOCK
qm.connectWithOptions(name, cd=cd, opts=connectOptions)


if __name__ == '__main__':
    handle = pymqi.MessageHandle(qmgr=qm)
    handle.properties.set(b'key', 'val')
```

Change the 3 variables at the top as needed for a particular configuration when running.

**Comment by dsuch** [comment]
Hello @KYDronePilot,

does changing data types of keys and values from bytes to unicode, or the other way around, make any difference?

Making PyMQI handle both types transparently was one of changes in 1.9.3.

Regards.

**Comment by SeyfSV** [comment]
@KYDronePilot , I checked your snippet on Ubuntu 18.04 VM with MQ client 9.1.3.0 - issue does not occur :(

**Comment by KYDronePilot** [comment]
@dsuch I tried every possible combination of bytes and unicode and it still did not work.

@SeyfSV I tried the snippet on an Ubuntu 18.04 VM as well, to ensure the issue wasn't being caused by the server. It ran no problem.

I believe the issue might have to do with the version of Python. I was able to get it working in an Ubuntu 18.04 container, which has Python 3.7.5 installed, while the Python version where I had the original issue was 3.7.3 (in Ubuntu 19.04). Ubuntu 19.10 appeared to work as well, which also has Python version 3.7.5.

I won't be able to do any further testing until next week, just fyi.

**Comment by SeyfSV** [comment]
@KYDronePilot Do you really need Python 3.7.3 in your project?

**Comment by KYDronePilot** [comment]
@SeyfSV No, it was just a side-effect of using Ubuntu 19.04. There should be no issues moving to a later Python version.

I'll leave it up to you all to decide whether this warrants a fix or not, since there are no issues with 3.7.5+.

**Comment by SeyfSV** [comment]
Ok. I close it.
